tests/unittests/library/test_nsx_common.py

```diff
@@ -628,6 +637,151 @@ def test_nsx_read(self):
             'c'
         )
 
+    @pytest.mark.internal
+    @pytest.mark.unit
+    def test_nsx_search(self):
+        """Check nsx_common.nsx_search func"""
+        self._regen_ctx()
+        client_session = mock.Mock()
+
+        # we raise error?
+        client_session.read = mock.Mock(return_value={
+            'body': {}, 'status': 300
+        })
+        with self.assertRaises(cfy_exc.NonRecoverableError):
```

Comment:
It would probably also be good to confirm that the error message contains pertinent information.

e.g.
```
try:
    common.nsx_search(client_session, 'body/a', 'b', 'secret',
                       uri_parameters={'scopeId': 'scopeId'})
    raise AssertionError('NSX search should not have succeeded with bad HTTP result')
except cfy_exc.NonRecoverableError as err:
    self.assertIn('not found')
    self.assertIn('HTTP request failure')
```

(I'm not sure if assertIn exists, but hopefully my suggestion is clear?)